djangoessentials/djangoessentials.py

```diff
@@ -15,3 +17,38 @@ class MyS3Storage(S3Boto3Storage):
     location = 'media/'  # S3'te files alt dir
     file_overwrite = False  # if have a same name about the file cant overwrite
     default_acl = 'public-read' # just can read permission
+
+
+
+#CustomUser Model
+class CustomUserManager(BaseUserManager):
+    def create_user(self, username, password=None, **extra_fields):
+        if not username:
+            raise ValueError('The Username field must be set')
+        user = self.model(username=username, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+    
+    def create_superuser(self, username, password=None, **extra_fields):
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+        return self.create_user(username, password, **extra_fields)
+
+
+class CustomUser(AbstractBaseUser, PermissionsMixin):
```

Comment:
its will be great to add this thing too. `class CustomUser(AbstractBaseUser, PermissionsMixin, TimeBasedStampModel):` @crabcatcher0

Reply:
Thank you for guiding.